File: udemy/java/masterclass/STS4/_6.0-OOP/src/com/timbuchalka/_77_classes/ex30/Person.java
package com.timbuchalka._77_classes.ex30;

public class Person {

	private String firstName;
	private String lastName;
	private int age;
	
	public String getFullName() {
		String fullname = "";
		if (!firstName.isEmpty() && !lastName.isEmpty()) {
			fullname += firstName + " " + lastName;
		} else if (firstName.isEmpty()) {
			fullname += lastName;
		} else if (lastName.isEmpty()) {
			fullname += firstName;
		}
		
		return fullname;
	}
	
	public boolean isTeen() {
		return (age > 12 && age < 20);
	}

	public String getFirstName() {
		return firstName;
	}

	public void setFirstName(String firstName) {
		this.firstName = firstName;
	}

	public String getLastName() {
		return lastName;
	}

	public void setLastName(String lastName) {
		this.lastName = lastName;
	}

	public int getAge() {
		return age;
	}

	public void setAge(int age) { 
		this.age = (age < 0 || age > 100)? 0 : age;
	}
}

File: wozuforlife/csharp/VSC/05_Objects/handsonl5/solutions/Person.cs
using System;

namespace handsonl5.solutions
{

    class Person
    {
        public Person() { }

        public Person(string firstName, string lastName, int age)
        {
            this.firstName = firstName;
            this.lastName = lastName;
            this.age = 0;

            if (age >= 0)
            {
                this.age = age;
            }
            else
            {
                Console.WriteLine(
                    "Attempted to set age to an invalid value. Set age to 0 instead.");
            }
        }

        private string firstName;
        public string FirstName
        {

            get { return this.firstName; }
            set { this.firstName = value; }
        }

        private string lastName;
        public string LastName
        {

            get { return this.lastName; }
            set { this.lastName = value; }
        }

        private int age;
        public int Age
        {

            get { return this.age; }
            set { this.age = value; }
        }
    }
}